vendor/parse-wyc/lib/annot.mli

```diff
@@ -1,15 +1,17 @@
-module type Annotated = sig
-  type t
+module Exp : sig
+  val mk : unit -> Migrate_ast.Parsetree.expression
 
-  val mk : unit -> t
-
-  val is_generated : t -> bool
+  val is_generated : Ppxlib.Parsetree.expression -> bool
 end
 
-open Migrate_ast.Parsetree
+module Attr : sig
+  val mk : unit -> Migrate_ast.Parsetree.attribute
 
-module Exp : Annotated with type t = expression
+  val is_generated : Ppxlib.Parsetree.attribute -> bool
+end
 
-module Attr : Annotated with type t = attribute
+module Class_exp : sig
+  val mk : unit -> Migrate_ast.Parsetree.class_expr
 
-module Class_exp : Annotated with type t = class_expr
+  val is_generated : Ppxlib.Parsetree.class_expr -> bool
+end
```

Comment:
It's both. The use of ppxlib makes this part more complicated than it used to be, so I'll try to summarize it (and that can go in a module comment or in a commit message):

The modified parser inserts generated nodes in places that would correspond to parse errors. For example, if it determines that some sequence of tokens can only form an expression (but it's missing some tokens to do so in a valid way), it will emit `[%merlin.hole]` as an expression. Later, an ast mapper walks the AST and needs to determine if an expression is generated or not.

This is simple with OMP, but with ppxlib we use two different AST versions: the parser uses 4.08 (from OMP) and the mapper uses 4.11 (at the time; more generally, the one ppxlib uses). So in the `Annot` module, the `mk` functions return a 4.08 AST (they are called by the parser), but the `is_generated` functions expect a 4.11 one (they are called by a mapper after migration).

Reply:
I added that to the commit message